### ISSUE TITLE
Style/RedundantSelf confused when `extending` a scope

### ISSUE THREAD

**Issue by nikolai-b** [rubocop-hq/rubocop#4321]
## Expected behavior

```ruby
module MyClassQuery
  def initialize(scope: MyClass.all)
    @scope = scope.extending(Scopes)
  end

  module Scopes
    def active
      self.open.where(active: true)
    end
  end
end

class MyClass  < ActiveRecord::Base
  scope :open, -> { where(open: true) }
end
```

The `self.open` should not trigger the redundant self cop.  Without the `self` ruby calls https://ruby-doc.org/core-2.1.4/File.html#method-c-open, with the `self` it calls `MyClass.instance_method(:open)` which is the intended behavior.

## Actual behavior

The cop is triggered

## Steps to reproduce the problem

See Expected behavior

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-linux)
```

**Comment by bbatsov** [comment]
You found a pretty funky edge case - seems we need to add some extra check that the method name is not some `Kernel` method.
